src/inmanta/ast/type.py

```diff
@@ -94,6 +106,46 @@ def with_base_type(self, base_type: "Type") -> "Type":
         """
         return base_type
 
+    def corresponds_to(self, type: "Type") -> bool:
+        """
+        Determine if the given 'type' is a good approximation to `self` given that `type` is derived from a python type
+
+        Intended specifically for type correspondence for dataclasses.
+        This brings specific assumptions
+        - `self` is the native inmanta type
+        - `type` is translated form the python domain
+        - 'Any' means: trust me, it will be fine (i.e. always corresponds)
```

Comment:
May become clear from the rest of the review, but does this mean that we also accept python's `object` without question? Iirc we translate it to `Any` internally.

Reply:
I have no strong objection. It's just very wide on the Python side. But there is always still the runtime validation (right?) so it's not too bad.

Reply:
on the model side: yes